### PR TITLE
chore: adds fianu plugin

### PR DESCRIPTION
**Pull request by karanadep-dexcom**
## Hey, I just made a Pull Request!

This PR adds the Fianu Reports plugin to the Plugin Marketplace.

This plugin was added in PR https://github.com/backstage/backstage/pull/19932 by @karanadep-dexcom

Screenshot of Fianu page:
Screen Shot 2023-09-14 at 12 21 42 PM

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/fianu/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# [Fianu](https://www.fianu.io/)
```
Comment (jhaals) [inline plugins/fianu/README.md:1]: Fianu needs to be added to the `.github/vale/Vocab/Backstage/accept.txt`

File: plugins/fianu/src/api/FianuApiClient.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ConfigApi } from '@backstage/core-plugin-api';
+import { FianuApi } from './FianuApi';
+
+export class FianuApiClient implements FianuApi {
```
Comment (jhaals) [inline plugins/fianu/src/api/FianuApiClient.ts:19]: Not sure that this warrants its own API at the moment given that it only concatenates a string from config which does not need to be async at the moment. Could probably be moved to the component instead.

File: plugins/fianu/src/components/annotationHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+
+/** @public */
+export const FIANU_BADGE_ANNOTATION = 'dexcomdev.com/fianu-badge';
```
Comment (jhaals) [inline plugins/fianu/src/components/annotationHelpers.ts:19]: The value of the annotation should be related to the plugin and it does not look like dexcomdev is related.

```suggestion
export const FIANU_BADGE_ANNOTATION = 'fianu.io/badge';
```

I'm not sure what a badge represent here, is it an id of some sort? In. that case badgeId would be preferred.

File: plugins/fianu/src/components/FianuComponent/FianuComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    <iframe
+      style={{
+        display: 'table',
+        width: '100%',
+        height: '100%',
+        border: '0px'
+      }}
+      title="Fianu Badge"
+      src={value}
+    />
```
Comment (jhaals) [inline plugins/fianu/src/components/FianuComponent/FianuComponent.tsx:39]: Would be great to see a screenshot how this displays.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the contributions, I've added some comments. Please note that all commits need to be DCO signed in order to be merged in.

File: plugins/fianu/src/components/annotationHelpers.ts (hunk 3 above)
Comment (karanadep-dexcom) [reply plugins/fianu/src/components/annotationHelpers.ts:19]: yes makes sense. I was planning on changing this.

File: plugins/fianu/README.md (hunk 1 above)
Comment (karanadep-dexcom) [reply plugins/fianu/README.md:1]: done

File: plugins/fianu/src/api/FianuApiClient.ts (hunk 2 above)
Comment (karanadep-dexcom) [reply plugins/fianu/src/api/FianuApiClient.ts:19]: I am planning to add some more functionality in future. so would like to keep it as separate API so we have pattern in place.

File: plugins/fianu/src/components/FianuComponent/FianuComponent.tsx (hunk 4 above)
Comment (karanadep-dexcom) [reply plugins/fianu/src/components/FianuComponent/FianuComponent.tsx:39]: done

**Comment by karanadep-dexcom** [comment]
@jhaals - Can you comment on GHA errors I am seeing ?

**Comment by jhaals** [comment]
> @jhaals - Can you comment on GHA errors I am seeing ?

Which one specifically, there are several where dependencies aren't added to package.json

```
     @backstage/plugin-catalog-react must be declared in dependencies of plugins/fianu/package.json, run 'yarn --cwd plugins/fianu add @backstage/plugin-catalog-react' from the project root
```

There might also need to re-generate your API report with `yarn build:api-reports`

File: plugins/fianu/src/api/FianuApiClient.ts (hunk 2 above)
Comment (jhaals) [reply plugins/fianu/src/api/FianuApiClient.ts:19]: Sounds good to me 👍

**Pull request closed without merging**